File: app/src/main/java/fr/skyost/timetable/lesson/ringer/RingerModeStateChangeReceiver.java
package fr.skyost.timetable.lesson.ringer;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.media.AudioManager;
import android.os.AsyncTask;

public class RingerModeStateChangeReceiver extends BroadcastReceiver {

	@Override
	public void onReceive(final Context context, final Intent intent) {
		AsyncTask.execute(() -> {
			if(LessonModeManager.isEnabled(context) && LessonModeManager.inLesson(context)) {
				final AudioManager manager = (AudioManager)context.getSystemService(Context.AUDIO_SERVICE);
				if(manager == null || LessonModeManager.getPreferenceMode(context) == manager.getRingerMode()) {
					return;
				}

				LessonModeManager.disable(context);
			}
		});
	}

}

File: android/app/src/main/kotlin/fr/skyost/timetable/ringer/RingerModeStateChangeReceiver.kt
package fr.skyost.timetable.ringer

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.media.AudioManager
import android.os.AsyncTask

class RingerModeStateChangeReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        AsyncTask.execute {
            if (LessonModeManager.isEnabled(context) && LessonModeManager.inLesson(context)) {
                val manager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
                if (manager == null || LessonModeManager.getPreferenceMode(context) === manager.ringerMode) {
                    return@execute
                }
                LessonModeManager.disable(context)
            }
        }
    }
}